lib/pages/Mainpage.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:MobileOne/localization/localization.dart';
+import 'package:MobileOne/pages/lists.dart';
+import 'package:MobileOne/pages/loyalty_card.dart';
+import 'package:MobileOne/pages/profile.dart';
+import 'package:MobileOne/pages/share.dart';
+import 'package:flutter/material.dart';
+
+class MainPage extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return MainPageState();
+  }
+}
+
+class MainPageState extends State<MainPage> {
+
+  int currentTab = 0;
+  final List<Widget> screens = [
+    LoyaltyCards(),
+    Lists(),
+    Share(),
+    Profile()
+  ];
+
+  final List _centerIcons =[
+    Icons.add,
+    Icons.add,
+    Icons.share,
+    Icons.camera,
+  ];
+
+    final List _centerText =[
+    'center_icon_loyaltycards',
+    'center_icon_newlist',
+    'center_icon_share',
+    'center_icon_profile'
+  ];
+
+  Widget currentScreen = Lists();
+  final PageStorageBucket bucket = PageStorageBucket();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: Text("XPEHO"),
+        ),
+        body:PageStorage(bucket: bucket, child: currentScreen),
+        floatingActionButton: FloatingActionButton(
+          child:Icon(_centerIcons[currentTab]),
+          backgroundColor: Colors.orangeAccent,
+          onPressed: (){},
+        ),
+        floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
+        bottomNavigationBar: BottomAppBar(
+          shape: CircularNotchedRectangle(),
+          child:Container(
+            height: 60,
+            child:Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+              Row(
+                children:<Widget>[
+                  MaterialButton(
+                    key:Key("Cards"),
+                    minWidth:40,
+                    onPressed:(){
+                      setState((){
+                        currentScreen = LoyaltyCards();
+                        currentTab = 0;
+                      });
+                    },
+                    child:Column(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children:<Widget>[
+                        Icon(Icons.loyalty),
+                        Text(getString(context,'loyalty_cards'))
+                      ]
+                    )
+                  ),
+                  MaterialButton(
+                     key:Key("Lists"),
+                    minWidth:40,
+                    onPressed:(){
+                      setState((){
+                        currentScreen = Lists();
+                        currentTab = 1;
```

Comment:
Can you please create constants to call each page ? 0, 1, 2, 4 is kind of "magic numbers" and we don't understand what's the real signification.
Create constants like const LISTS_PAGE = 1;